rfc/0018-certificate-handling.md

```diff
@@ -82,174 +86,199 @@ webhooks:
+CertManager.
 
-Each `ClusterAdmissionPolicy` and `AdmissionPolicy` has a dedicated `(Validating|Mutating)WebhookConfiguration`
-object defined. For all of them, the `clientConfig.caBundle` contains the certificate of
-the CA generated by our controller.
+Each `ClusterAdmissionPolicy` and `AdmissionPolicy` has a dedicated
+`(Validating|Mutating)WebhookConfiguration` object defined. For all of them,
+the `clientConfig.caBundle` contains the certificate of the CA generated by our
+controller.
 
 ## Proposed Design
 
-We would like to get rid of CertManager. To do that we need to change how certificates are
-managed for the `kubewarden-controller`.
-Going forward, the controller will also take care of generating the certificate used by the kubewarden-controller.
-This certificate is going to be signed by the CA which is already created by the controller.
-
-```mermaid
-flowchart TD
-    A[KW Internal CA]
-    A -->B(KW controller)
-    A -->C(Policy Server #1)
-    A -->D(Policy Server #2)
-```
+We would like to get rid of CertManager. To do that we need to change how
+certificates are managed for the `kubewarden-controller`. Going forward, the
+Helm chart installation and  the controller will take care of generating the
+certificate used by the `kubewarden-controller`. 
 
 ## Possible Scenarios
 
-This section aims to cover the different scenarios. For each one of them we start by defining an event
-and then we outline the different actions that have to be done.
+This section aims to cover the different scenarios. For each one of them we
+start by defining an event and then we outline the different actions that have
+to be done.
 
 ### Kubewarden Fresh Install
 
-The Kubewarden stack has just been installed. The following actions have to be done:
+The Kubewarden stack has just been installed. The following actions have to be
+done:
 
-- Generate Certificate Authority
-- Generate certificate for `kubewarden-controller` and sign it with our CA
-- Start the Kubewarden controller HTTPS server using this certificate
-- Register the webhook against the Kubernetes API, using the certificate of our CA inside of
-  `clientConfig.caBundle`
+- Helm chart generates Certificate Authority(CA) and store it in a secret
+- Helm chart generates cerficate for `kubewarden-controller`, sign it with the
+  CA, and store it in a secret
+- Helm chart uses the CA in the `caBundle` configuration in webhooks to
+  validate CRDs pointing to the `kubewarden-controller`
+- Start the Kubewarden controller HTTPS server using this certificate. The
+  certificate is mounted inside of the Pod
 
 ### New Policy Server Defined
 
 A new `PolicyServer` is defined, the following actions have to be done:
 
 - Generate new certificate, sign it with our CA
-- Create Policy Server Deployment, ensure the certificate is mounted inside of the Pods
+- Create Policy Server Deployment, ensure the certificate is mounted inside of
```

Comment:
We don't ensure that a certificate is mounted anymore. We leverage an auto-reload mechanism since the secret volume will eventually mounted/updated by Kubernetes.